docs/codeql/codeql-language-guides/abstract-syntax-tree-classes-for-working-with-ruby-programs.rst

```diff
@@ -414,6 +418,32 @@ All classes in this subsection are subclasses of VariableAccess_.
 | ``self``                   | SelfVariableReadAccess_      | VariableReadAccess_, SelfVariableAccess_      |                  |
 +----------------------------+------------------------------+-----------------------------------------------+------------------+
 
+Desugaring
```

Comment:
Do we expect the target audience for this section to be confident of the meaning of "desugaring"? Would it be better to talk about simplifying and expanding the simplified syntax, or something along those lines?

Reply:
That is a good point. Not every user will be familiar with the terminology.

Reply:
True - I think we should mention the term "desugaring" as part of the text as we have a `getDesugared` predicate, but probably with a description of what it means. I think that "syntactic sugar" is a commonly understood term, though?